pcdsdevices/happi/containers.py

```diff
@@ -23,6 +23,24 @@ class LCLSItem(OphydItem):
                       enforce=re.compile(r'[A-Z0-9]{3}$|[A-Z][0-9]S[0-9]{2}$'))
     lightpath = EntryInfo("If the device should be included in the "
                           "LCLS Lightpath", enforce=bool, default=False)
+    ioc_engineer = EntryInfo(('Engineer for the IOC. Used to build IOC '
+                             'configs.'),
+                             optional=True, enforce=str)
+    ioc_location = EntryInfo('Loction of the IOC. Used to build IOC configs.',
+                             optional=True, enforce=str)
+    ioc_hutch = EntryInfo(('Hutch the IOC will be used in. Used to build IOC '
+                           'configs.'), optional=True, enforce=str)
+    ioc_release = EntryInfo(('Full path to IOC release directory. Used to '
+                            'build IOC configs.'),
+                            optional=True, enforce=str)
+    ioc_arch = EntryInfo(('The IOC host architecture. Used to build IOC '
+                         'configs.'), optional=True, enforce=str)
+    ioc_name = EntryInfo('The name of the device IOC. Used to build the IOC.',
+                         optional=True, enforce=str)
+    ioc_type = EntryInfo(('The type of device IOC. Useful when multiple '
+                          'device classes can occupy the same controller. Can '
+                          'be used to tell higher level code how to interpret '
+                          'the ioc data.'), optional=True, enforce=str)
```

Comment:
I've noticed that all of the `EntryInfo`s you've added are optional, even on your laser-specific subclasses. Is this what you want for things that you're building IOCs for? Another way to formulate this would have been a `LCLSAutoIOC` subclass with these `ioc_hutch`, etc. fields as `optional=False`, then subclass that for things like the `Elliptic`. I'm curious as to your thoughts on this.

Reply:
My thinking there was that the laser classes I'm writing may still be used for Happi database entry, but without including the IOC generation data. There's talk about rolling out similar laser controls and diagnostics to the FEH soon, where we may be doing things more manually than I hope to do for the MODS systems in the NEH. My thinking was that if you're using the IOC data, you will know that it's required and won't need the optional flag to remind you. Additionally, the mods_deploy code is using schema to do some more checking on the data, so if something is missing the mods_deploy code will let you know. 

I considered creating a laser IOC  parent class as you suggest, but thought that if this becomes more generally useful that it would be nice to have some of the data in a common class. I'm open to moving things to a laser parent class if you guys like that better.

Reply:
Makes sense to me :+1: